### ISSUE TITLE
feat: PrestoDB backend

### ISSUE THREAD

**Issue by stereoF** [ibis-project/ibis#7324]
### Is your feature request related to a problem?

Yes, while working on data-related tasks, I often find myself using PrestoDB due to its capabilities. However, there's currently no direct support for PrestoDB in Ibis, which means I have to switch between tools or write raw SQL, making the process less efficient.

### Describe the solution you'd like

I'd like Ibis to have a native support for PrestoDB, allowing users to leverage Ibis's expressive API to interact with PrestoDB seamlessly.

Key benefits would include:
- **Unified Querying**: Transition between different backends with ease, using the same Ibis API.
- **Integration**: Benefit from PrestoDB's capability of querying data from multiple sources without switching tools.
- **Performance**: Utilize both Ibis's expressiveness and PrestoDB's performance advantages for efficient data processing.

An added support for PrestoDB would certainly elevate the usability of Ibis for many data professionals who rely on PrestoDB for their tasks.

### What version of ibis are you running?

latest

### What backend(s) are you using, if any?

PrestoDB

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by cpcloud** [comment]
Hi @stereoF 👋🏻! Welcome to Ibis, and thanks for the issue!

+1 on a PrestoDB backend.

Any chance you can comment on the setting in which you're using PrestoDB? We love to hear about how people are using (or would like to use!) Ibis.

In any case, regarding implementation I think we can probably share a lot of the Trino backend's expression translation rules, and keep the differences mainly to connecting to the database, with perhaps a few PrestoDB specific expression translation rules.

**Comment by stereoF** [comment]
> Hi @stereoF 👋🏻! Welcome to Ibis, and thanks for the issue!
> 
> +1 on a PrestoDB backend.
> 
> Any chance you can comment on the setting in which you're using PrestoDB? We love to hear about how people are using (or would like to use!) Ibis.
> 
> In any case, regarding implementation I think we can probably share a lot of the Trino backend's expression translation rules, and keep the differences mainly to connecting to the database, with perhaps a few PrestoDB specific expression translation rules.

Hello,
Firstly, I want to extend my gratitude to the Ibis community for your support. Our company has implemented an OLAP platform with its persistence layer on hdfs and the query engine being Presto. Typically, the OLAP platform is geared towards agile analysis, and its table structure is based on an event-driven model. As we delve deeper into machine learning modeling, we often need to transition from this event-based structure to a wide-table feature construction.

Back between 2019 and 2020, I worked on a similar OLAP platform during my tenure at Tencent. I developed some generic analysis model tools, and at that time, the query engine was Impala. My approach was to dynamically concatenate SQL, which unfortunately was not conducive to code encapsulation, modularization, and future maintenance.

In my pursuit of better code encapsulation and to decouple different parts of logic, I was initially inclined to use PySpark. However, when PySpark connects to Presto via JDBC, if we use the dataframe interface, the aggregation operations run on Spark. This doesn't harness the full power of Presto, leading to slow performances. On the other hand, if we use Spark's SQL interface, aggregation is processed on Presto. But in doing so, we lose the original intent of using Spark - which is better code encapsulation and the decoupling of different processes.

The dataframe interface of Ibis and its feature of lazy computation perfectly align with my needs. In fact, back in 2019, I was on the hunt for such a tool. Sadly, I didn't come across Ibis at that time and even contemplated creating a set on my own.

On a side note, today I successfully connected to the Presto engine using Ibis's Trino connector and so far, everything runs smoothly without any issues. However, using the Trino connector to connect to the Presto engine always has inherent risks. Hence, direct support for Presto would be ideal.
